Source/Randomizer.cpp

```diff
@@ -109,6 +111,14 @@ int find(const std::vector<T> &data, T search, size_t startIndex = 0) {
     throw std::exception("Couldn't find value in data!");
 }
 
+std::vector<int> copyWithoutElements(const std::vector<int>& input, const std::set<int>& toRemove) {
+  std::vector<int> result = input;
+  result.erase(std::remove_if(std::begin(result), std::end(result), [&toRemove](int val) {
+      return toRemove.count(val);
+  }), std::end(result));
+  return result;
+}
```

Comment:
Good idea, but the implementation here is a bit odd.
- Why not use an in/out variable? We don't need the input list to be immutable, and copying is potentially slow.
- There's a function called `std::remove_copy_if` which I feel like is more suited to the job, if you keep this orchestration.
- As wonderful as std::algorithm is, this code is now very hard to read (and reason about). Why not just use a simple for loop?
```cpp
for (auto& it : input) {
    if (toRemove.find(it) != toRemove.end()) it = input.remove(it);
}
```

Reply:
I think we do want to do a copy, because otherwise we're modifying the original set of panels, and that modification will be carried forward if the player later unchecks "double randomizer mode".
 
The code sample doesn't work because vector doesn't have a remove-by-value method, and even if it did, we'd be invalidating the iterator that the for loop is using. If we want to avoid using algorithm, I think the simplest thing might be something like this:

```c++
std::vector<int> result;
for (int val : input) {
    if (toRemove.find(val) == toRemove.end()) {
        result.push_back(val);
    }
}
```

Reply:
Whoops, I confused `remove` with `erase` again. Anyways, that's why it's `it = input.erase(it);`, erase returns a new iterator which will work safely with the for loop.

That said, if we're making a copy, that's not relevat.
Your suggestion looks good, just call `result.reserve()` to avoid extra resizing.

Reply:
Sounds good to me!